zilliqa/src/consensus.rs

```diff
@@ -600,6 +618,11 @@ impl Consensus {
             return Ok(None);
         }
 
+        if block.view() <= head_block.header.view {
```

Comment:
Some of the lower down logic relies on making sure the block is definetly from a higher view (reverting when forking).